src/UpdateCounterMigrationBundle/Tests/Migrator/UpdateCounterVersionMigratorTest.php

```diff
@@ -0,0 +1,321 @@
+<?php
+
+/*
+ * This file is part of the Chameleon System (https://www.chameleonsystem.com).
+ *
+ * (c) ESONO AG (https://www.esono.de)
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ChameleonSystem\UpdateCounterMigrationBundle\Tests\Migrator;
+
+use ChameleonSystem\UpdateCounterMigrationBundle\DataAccess\BundleDataAccessInterface;
+use ChameleonSystem\UpdateCounterMigrationBundle\DataAccess\CounterMigrationDataAccessInterface;
+use ChameleonSystem\UpdateCounterMigrationBundle\Exception\InvalidMigrationCounterException;
+use ChameleonSystem\UpdateCounterMigrationBundle\Migrator\UpdateCounterVersionMigrator;
+use PHPUnit\Framework\TestCase;
+use Prophecy\Argument;
+use Prophecy\Prophecy\ObjectProphecy;
+
+class UpdateCounterVersionMigratorTest extends TestCase
+{
+    /**
+     * @var UpdateCounterVersionMigrator
+     */
+    private $subject;
+    /**
+     * @var CounterMigrationDataAccessInterface|ObjectProphecy
+     */
+    private $counterMigrationDataAccessMock;
+    /**
+     * @var BundleDataAccessInterface|ObjectProphecy
+     */
+    private $bundleDataAccessMock;
+
+    protected function tearDown()
+    {
+        parent::tearDown();
+        $this->subject = null;
+        $this->counterMigrationDataAccessMock = null;
+        $this->bundleDataAccessMock = null;
+    }
+
+    /**
+     * @dataProvider provideDataForTestMigrateToVersionTwo
+     *
+     * @param array $oldCounters
+     * @param array $newCounters
+     */
+    public function testMigrateToVersionTwo(array $oldCounters, array $newCounters): void
+    {
+        $this->givenUpdateCounterVersionMigrator();
+        $this->givenBundlesAreRegistered();
```

Comment:
The migration checks for invalid counters and displays a list of invalid counters, so that the user can decide what to do with them. The different choices are outlined in the translation string "chameleon_system_core.cms_module_update.error_invalid_counter_names" (already part of 6.2.0 and outside this PR).

This check on invalid counters is tested in testMigrateToVersionTwoInvalidCounters. It ensures that no migration work is done unless there are no more invalid counters. 

I think it is OK to define a fixed set of registered bundles for all tests and reference invalid counters in the dedicated testMigrateToVersionTwoInvalidCounters.